### ISSUE TITLE
Oryx Build Gives Error if Optional Parameters are not set

### ISSUE THREAD

**Issue by raahmed** [Azure/static-web-apps#140]
According to the Static Web Apps tutorial, the app artifact location is not a necessary parameter for creating a static web-app:

![image](https://user-images.githubusercontent.com/25991359/91498992-906b7300-e875-11ea-8635-e6a93f0a8c9c.png)

However, if a user attempts to deploy a static web-app without this parameter (reference [file](https://github.com/raahmed/knboard/blob/master/.github/workflows/azure-static-web-apps-zealous-flower-01dc36d1e.yml))...

![image](https://user-images.githubusercontent.com/25991359/91499085-b98c0380-e875-11ea-98d1-a08a16e91acb.png)

[One](https://github.com/raahmed/knboard/runs/1038238506?check_suite_focus=true) of the CI builds shows an error:

![image](https://user-images.githubusercontent.com/25991359/91499047-a711ca00-e875-11ea-8f06-8394830bcb71.png)

**Comment by miwebst** [comment]
This looks to be a repeat of this: https://github.com/Azure/static-web-apps/issues/4

Basically this parameter is optional depending on your application type. If you aren't using a framework (i.e. pure html/css), then this is optional you dont need it. If you are using a framework (React in your case), you should provide this property.

Looking at your application it should be set to  `build `  as this is the output directory for your app.

cc @craigshoemaker

**Comment by craigshoemaker** [comment]
@raahmed - which tutorial were you reading?

**Comment by raahmed** [comment]
@craigshoemaker : This one : https://docs.microsoft.com/en-us/learn/modules/publish-app-service-static-web-app-api/3-static-web-apps

**Comment by craigshoemaker** [comment]
thanks, @raahmed!

@johnpapa this is in your Learn module. Can you take a look, please?

**Comment by johnpapa** [comment]
@craigshoemaker - i'm not sure how to re-word this. Based on what @miwebst  says above, I get the issue ... but I wonder if this is something we can produce a better error message on. Or better yet, make the service handle it if its blank.

Otherwise ... we could word it as 

> This setting is optional except when you use a web framework that expects output to be placed in a distribution folder. In those cases, enter the distribution folder in this box.

I'm torn

**Comment by miwebst** [comment]
Long-term we should be able to make this fully optional, the service will be able to do the required detection so this field wont be required even for frameworks. Today it is required because that detection is not in place. Would it make sense to reword that as something as simple as:

"This setting is required when using web frameworks that generate an output folder (i.e. React, Gatsby, etc.), but is not required if your app does not use any such framework."

**Comment by craigshoemaker** [comment]
@johnpapa - were you able to update the text in this unit?

@miwebst - Is this functionality addressed in the GA release by any chance?

**Comment by johnpapa** [comment]
yes - the Learn module uses the VS Code extension which skips over this

**Comment by anthonychu** [comment]
We don’t auto detect this right now.
